### PR TITLE
7 data preprocessing

### PR DESCRIPTION
**Pull request by PranjalDhamane**
Dear All,

I have completed preprocessing from my side but before that please review and if there are some suggestions or changes required do let me know. 

@Asifdotexe 
@str04 
@aditimane07 
@tanvisivaraj

### CODE REVIEW

File: code/notebook/issue-7-pd-data-processing.ipynb [hunk 1]
```diff
@@ -0,0 +1,929 @@
...
+       "(25499, 9)"
+      ]
+     },
+     "execution_count": 39,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df = df.query(\"genres not in ['Ecchi', 'Hentai', 'Erotica']\")\n",
```
Comment (Asifdotexe) [inline code/notebook/issue-7-pd-data-processing.ipynb:468]: Still there are instances of Ecchi and other mature content present

File: code/notebook/src/cleaning.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+from nltk import pos_tag
+nltk.download('punkt')
+nltk.download('stopwords')
+nltk.download('wordnet')
+nltk.download('averaged_perceptron_tagger')
+
+stemmer = PorterStemmer()
+lemmatizer = WordNetLemmatizer()
+
+def nlp_cleaning(text):
```
Comment (Asifdotexe) [inline code/notebook/src/cleaning.py:16]: Add the code for removing special character as well, the title contains some special characters in them.
all the non ASCII values should be removed, refer to the image below

in the title

image

File: code/notebook/src/cleaning.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    str: The cleaned text after applying the following operations:
+         
+    - Convert the text to lowercase.
+    - Remove URLs from the text.
+    - Remove non-ASCII characters.
+    - Remove numbers.
+    - Remove punctuation.
+    - Remove extra whitespaces.
+    """
+    if not isinstance(text, str):
```
Comment (Asifdotexe) [inline code/notebook/src/cleaning.py:33]: Can we do the following preprocessing on the episode column.

I have seen that the data has a lot of missing values but there is a pattern to it. 
Movies are usually just individual pieces of media and don't usually have episodes hence in that case, we do not have the episode count populated.

Can we do imputation on the records where we can find "movie" in the title, the value should be filled as 1

Comment (Asifdotexe) [reply code/notebook/src/cleaning.py:33]: After this, we can populate the rest of the missing values as -1

**Review by Asifdotexe — CHANGES_REQUESTED** [review]
Made some changes. please make sure you intregrate them

**Review by Asifdotexe — DISMISSED** [review]
Looks good to me

**Review by aditimane07 — DISMISSED** [review]

**Review by str04 — APPROVED** [review]
It looks good to me also.

**Comment by Asifdotexe** [comment]
The branch was being merged with main, refrain from doing that. All the changes needs to be initially pushed to #3 develop branch. 

The idea behind this is, we will be treating it as staging area for integrating the changes. As we are programming in different modules. 

Once we are able to run it on develop bug free / with minimal bugs then we will be pushing the code to main to conclude on release.

@aditimane07 
@str04 
@tanvisivaraj 
@PranjalDhamane 

While creating pull requests, make sure you are asking to merge on #3 develop branch instead of main

**Pull request merged**